recomp-verify.py

```diff
@@ -73,78 +78,196 @@ def verify_single_process(spec, cfg, cust, naive, verbose):
+                if content:  # Only print non-empty logs
+                    print(content)
+                else:
+                    print(f"\n{subdir}.log is empty.")
+        else:
+            print(f"\nLog file {subdir}.log does not exist in {subdir}.")
+
+def run_multi_verif_with_parallel(dest_dir, spec, cfg, verbose=False):
+    # Get the absolute path to recomp-verify.py using root_dir
+    script_path = os.path.join(root_dir, "recomp-verify.py")
 
-    # spawn a process for the naive map
-    f = partial(verify_capture_output, spec, cfg, "naive", "--naive")
-    futures.append( executor.submit(f) )
+    # Prepare the individual subdirectories for each verification case
+    subdirs = ["mono", "cust_1", "cust_2", "naive"]
+    
+    # Determine the original directory (the one containing the TLA and CFG files)
+    orig_dir = os.path.dirname(os.path.abspath(spec))
+    if not orig_dir:
+        orig_dir = os.getcwd()
 
-    # spawn a process for each recomp map
-    pdirs = ["mono", "cust_1", "cust_2"]
-    for pdir in pdirs:
-        f = partial(verify_capture_output, spec, cfg, pdir, "--cust", "custom_recomp.csv")
-        futures.append( executor.submit(f) )
+    # Define the path to no_invs.cfg
+    no_invs_cfg_path = os.path.join(root_dir, "no_invs.cfg")
 
-    done, not_done = concurrent.futures.wait(futures, return_when=concurrent.futures.FIRST_COMPLETED)
-    winner = done.pop()
-    output = winner.result()
-    #for future in not_done:
-        #future.cancel()
-    return output
+    # Loop through subdirs to create each directory and copy spec, cfg, and no_invs.cfg files
+    for subdir in subdirs:
+        subdir_path = os.path.join(dest_dir, subdir)
+        os.makedirs(subdir_path, exist_ok=True)
+        
+        # Copy the required files only if they don't already exist
+        shutil.copy(spec, os.path.join(subdir_path, os.path.basename(spec)))
+        shutil.copy(cfg, os.path.join(subdir_path, os.path.basename(cfg)))
+        if os.path.exists(no_invs_cfg_path):
+            shutil.copy(no_invs_cfg_path, os.path.join(subdir_path, "no_invs.cfg"))
+
+    # Define strategies and corresponding flags
+    strategy_flags = {
+        "mono": "--cust",
+        "cust_1": "--cust",
+        "cust_2": "--cust",
+        "naive": "--naive"
+    }
+
+    # Define file names for each corresponding .sh file
+    script_names = ["mono.sh", "cust_1.sh", "cust_2.sh", "naive.sh"]
+
+    # Generate each shell script in the desired format
+    for subdir, script_name in zip(subdirs, script_names):
+        # The strategy flag
+        flag = strategy_flags[subdir]
+
+        # Shell script content:
+        script_content = f"""#!/usr/bin/env bash
+echo "Running {script_name}"
+echo "Spec file is: {spec}"
+echo "CFG file is: {cfg}"
+echo "Original directory is: {orig_dir}"
+
+# Navigate to the target directory
+cd "./{subdir}" || {{
+    echo "Failed to cd into the target directory." >&2
+    exit 1
+}}
```

Comment:
please hide the creation of the script behind a function. for example:

create_script(script_name, spec, cfg, ...):
  return f"""#!/usr/bin/env bash
  ...